includes/ucf-social-common.php

```diff
@@ -222,18 +248,110 @@ function ucf_social_links_display_after( $atts ) {
 	}
 }
 
+/**
+* Display the content before the social feed
+* @author RJ Bruneel
+* @since 1.0.4
+* @return string
+**/
+if ( ! function_exists( 'ucf_social_feed_display_before' ) ) {
+	function ucf_social_feed_display_before( $atts ) {
+		ob_start();
+	?>
+		<aside class="ucf-social-feed">
+	<?php
+		echo ob_get_clean();
+	}
+}
+
+/**
+* Display the social feed content
+* @author RJ Bruneel
+* @since 1.0.4
+* @param array $atts | Assoc. array of shortcode options
+* @return string
+**/
+if ( ! function_exists( 'ucf_social_feed_display' ) ) {
+	function ucf_social_feed_display( $atts ) {
+		$atts = shortcode_atts( array(
+			'feed'       => '',
+			'container'  => 'ucf-social-feed',
+			'layout'     => 'waterfall',
+			'grid-width' => 320,
+			'grid-rows'  => 3
+		), $atts );
+
+		global $post;
+		if ( !$post ) { return; }  // back out if there's no post data to reference
+		$permalink = urlencode( get_permalink( $post->ID ) );
+
+		ob_start();
+	?>
+		<div id="<?php echo $atts['container']; ?>"></div>
+		<script type="text/javascript">
+			$(function() {
+			<?php if($atts['layout'] === 'waterfall') : ?>
```

Comment:
If we think there's any chance of us needing custom social wall layouts in future sites, we should try to implement per-layout display hooks (e.g. `ucf_social_display_feed_waterfall`) to make it easier to add and override layouts.  Otherwise, adding a new layout would involve overriding this hook by re-adding all this logic along with the custom layout(s), which is not super ideal.

@jmbarne3 @rjucf Thoughts?

Reply:
I think if we need it in the future we can add it then.

Reply:
I only mention it now because adding it later would be a breaking change.

That being said, I'm inclined to suggest leaving it as-is and suggesting we stick with a consistent set of layouts/styles across sites.